filer/models/filemodels.py

```diff
@@ -506,16 +511,43 @@ def __lt__(self, other):
 
     @property
     def actual_name(self):
+        clean_name = self.clean_actual_name
+        if not self.sha1:
+            try:
+                self.generate_sha1()
+            except Exception:
+                return clean_name
+        suffix = self.sha1[:10]
+        if '.' not in clean_name:
+            return '{}.{}'.format(clean_name, suffix)
+        tokens = self.clean_actual_name.split('.')
+        tokens.insert(len(tokens)-1, self.sha1[:10])
```

Comment:
Yes, this is right, thanks. The problem could also be for 2016.06.picture.jpg, so I've moved the hash to the begining. This should be ok for all the cases.